lib/dynflow/exporters/abstract.rb

```diff
@@ -0,0 +1,95 @@
+module Dynflow
+  module Exporters
+    class Abstract
+
+      attr_reader :index
+
+      def self.export_execution_plan_id(world, execution_plan_id, options = {})
+        self.new(world, options)
+          .add_id(execution_plan_id)
+          .finalize.index[execution_plan_id][:result]
+      end
+
+      def self.export_execution_plan(execution_plan, options = {})
+        self.new(execution_plan.world, options)
+          .add(execution_plan)
+          .finalize.index[execution_plan.id][:result]
+      end
+
+      # Expected workflow:
+      # Exporter.new.add(foo).add(bar).add(baz).finalize
+      def initialize(world = nil, options = {})
+        @world   = world
+        @options = options
+        @index   = {}
+      end
+
+      # Add execution plan to export
+      def add(execution_plan)
+        @index[execution_plan.id] = {
+          :plan   => execution_plan,
+          :result => nil
+        }
+        self
+      end
+
+      # Add execution plan to export by id
+      def add_id(execution_plan_id)
+        @index[execution_plan_id] = {
+          :plan   => nil,
```

Comment:
please explain in the comment, where the `:plan` will get filled in?, and the whole difference between `add` and `add_id`